### PR TITLE
Symmetrical Text Centering

### PR DESCRIPTION
**Pull request by bronsonavila**
I created a fork of `venn.js` so I could modify how text is centered for [ASR-111](https://a57product.atlassian.net/jira/software/projects/ASR/boards/36/backlog?selectedIssue=ASR-111).

Before pushing up code, I ran the following three scripts to ensure everything was compiled and passed the internal tests:
```
npm run pretest
npm run test
npm run prepublish
```

To validate, clone the repo and open `examples/styled_centre.html` to see an example of text centering.

#### Details

The main thing that needed modification was the value of the `x` property in the `ret` variable on L.416 of `src/diagram.js`. I set up an optional parameter to control how that value will be set. If the `symmetricalTextCentre` option is set to `true`, then `x` will have a value of 0 and the text will be symmetrically centered as follows:

Screen Shot 2019-11-11 at 12 15 14 PM

Otherwise, if the option is not specified, then the default centering method will be used:

Screen Shot 2019-11-11 at 12 12 59 PM

I don't really know why the change made to `ret` works. I just experimented with things that seemed relevant to the issue until they worked the way I wanted them to.

Also, I used the UK variant of "centre" rather than "center" because that's the spelling used by the original author.

### CODE REVIEW

**Review by nateluzod — APPROVED** [review]
Looks good. How did you generate the examples in the PR notes? Is that from our app? If this hasn't been pulled into our react app yet, can it be?

**Comment by nateluzod** [comment]
Also, please wait for Evan's approval to merge in, just to be safe.

**Comment by bronsonavila** [comment]
@nateluzod I did not generate the examples. They are from the [original repo](https://github.com/benfred/venn.js/tree/master/examples). I just made a copy of the [styled.html](https://github.com/benfred/venn.js/blob/master/examples/styled.html) example (as `styled_centre.html`), and added the `symmetricalTextCentre` property to each Venn diagram in the file. 

I'll wait for a review from @evantbyrne before merging anything in.

**Review by evantbyrne — APPROVED** [review]
Well done! I'm not sure what ret means in this context either (maybe short for return?), but the [Nelder-Mead method](https://en.wikipedia.org/wiki/Nelder%E2%80%93Mead_method) seems relevant so I think this is right place for that check.

**Pull request merged**
